src/graphics/program-lib/programs/lit-shader.js

```diff
@@ -65,19 +66,23 @@ class LitShader {
         if (options.chunks) {
             this.chunks = {};
 
-            for (const p in shaderChunks) {
-                if (shaderChunks.hasOwnProperty(p)) {
-                    if (options.chunks[p]) {
-                        const chunk = options.chunks[p];
-                        for (const a in builtinAttributes) {
-                            if (builtinAttributes.hasOwnProperty(a) && chunk.indexOf(a) >= 0) {
-                                this.attributes[a] = builtinAttributes[a];
-                            }
+            const userChunks = options.chunks;
+
+            // #if _DEBUG
+            validateUserChunks(options.chunks);
+            // #endif
+
+            for (const chunkName in shaderChunks) {
+                if (userChunks[chunkName]) {
```

Comment:
Use `hasOwnProperty` for this check instead. Users can define an empty string as a chunk.

Reply:
Done, thanks